### PR TITLE
[Xero] [Export Flow] Create the Bank account select page

### PR DESCRIPTION
**Pull request by mananjadhav**
### Fixed Issues

$ https://github.com/Expensify/App/issues/39745
PROPOSAL:

### Tests

**Precondition:**

Workspace with Xero connection and has Bank accounts.

**Test steps**

1. Go to Workspace > Accounting > Export > Xero bank account
2. Verify that you see the accounts from the Xero Bank accounts list.
3. Click on any bank account.
4. Verify It displays the correct bank account name under Xero bank account in Export page.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as Test Steps

### QA Steps

Same as Test Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/3069065/cba72f2d-a989-4738-a2a8-c3e2eaf7a0e6

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/3069065/7622c6f4-f65b-4490-ac46-df0909a0ccd6
</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/3069065/35e1f115-f3e9-49bc-b791-45e9a4f17f62

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/3069065/eb931a7d-1e66-4794-b067-14d220c0e10e

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/3069065/d51b10d8-397a-4c9c-af62-8544f0e89529

web-xero-bank-account-select-es

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/3069065/5a0ceabb-2f14-43af-8392-1beb916bfd99

</details>

### CODE REVIEW

**Comment by lakchote** [comment]
@pecanoro could we please have the spanish translation for this:

```
xeroBankAccountDescription: 'Select the bank account expenses will be posted to as bank transactions.',
```

Link to the line in the PR: https://github.com/Expensify/App/pull/41916/files#diff-472528060691374654a2e2e866547d385eec3849efddca38cbfca1e5b3ba47d5R2079

**Comment by mananjadhav** [comment]
@ishpaul777 You'll need to enable the `Accounting` and `Xero` beta for this. You can check the steps [here](https://github.com/Expensify/App/pull/40277#issue-2245622576).

File: src/languages/es.ts [hunk 1]
```diff
@@ -2076,6 +2076,7 @@ export default {
                 'Cada gasto exportado se contabiliza como una transacción bancaria en la cuenta bancaria de Xero que selecciones a continuación. Las fechas de las transacciones coincidirán con las fechas de el extracto bancario.',
             bankTransactions: 'Transacciones bancarias',
             xeroBankAccount: 'Cuenta bancaria de Xero',
+            xeroBankAccountDescription: 'Select the bank account expenses will be posted to as bank transactions.',
```
Comment (pecanoro) [inline src/languages/es.ts:2079]:
```suggestion
            xeroBankAccountDescription: 'Selecciona la cuenta bancaria en la que se aparecerán los gastos como transacciones bancarias.',
```

**Review by pecanoro — CHANGES_REQUESTED** [review]

**Comment by mananjadhav** [comment]
@ishpaul777 Did you get the access working? Need to move this faster along.

File: src/languages/es.ts (hunk 1 above)
Comment (mananjadhav) [reply src/languages/es.ts:2079]: I'll update this in an hour.

**Comment by mananjadhav** [comment]
@pecanoro The Spanish translation is updated (screenshot added to the PR body as well).

web-xero-bank-account-select-es

File: src/languages/es.ts (hunk 1 above)
Comment (mananjadhav) [reply src/languages/es.ts]: Done.

**Comment by mananjadhav** [comment]
Asking @hungvu193 to review as per [this](https://expensify.slack.com/archives/C036QM0SLJK/p1715338321218469?thread_ts=1715336564.073279&cid=C036QM0SLJK). Need this for the Xero adhoc build.

**Comment by mananjadhav** [comment]
> Hey @mananjadhav, Sorry for delay, I was not able to priortize yesterday, will try the steps now let you know if i am able to setup in few minutes 🙇

Thanks for responding. Just asked @hungvu193. They already have the setup as well as the context.

**Comment by mananjadhav** [comment]
No. Means a Xero account with Bank names.

File: src/pages/workspace/accounting/xero/export/XeroBankAccountSelectPage.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    const xeroSelectorOptions = useMemo<SelectorType[]>(
+        () =>
+            (bankAccounts ?? []).map(({id, name}) => ({
+                value: id,
+                text: name,
+                keyForList: id,
+                isSelected: nonReimbursableAccount === id,
+            })),
+        [nonReimbursableAccount, bankAccounts],
+    );
```
[the remark this answers is not in the document]
Comment (mananjadhav) [reply src/pages/workspace/accounting/xero/export/XeroBankAccountSelectPage.tsx:34]: @lakchote Can confirm this. We didn't add it in other screens so I've kept it as is.

File: src/pages/workspace/accounting/xero/export/XeroBankAccountSelectPage.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            <View style={[styles.pb2, styles.ph5]}>
+                <Text style={[styles.pb5, styles.textNormal]}>{translate('workspace.xero.xeroBankAccountDescription')}</Text>
+            </View>
+        ),
+        [translate, styles.pb2, styles.ph5, styles.pb5, styles.textNormal],
+    );
+
+    const initiallyFocusedOptionKey = useMemo(() => xeroSelectorOptions?.find((mode) => mode.isSelected)?.keyForList, [xeroSelectorOptions]);
+
+    const updateMode = useCallback(
```
Comment (hungvu193) [inline src/pages/workspace/accounting/xero/export/XeroBankAccountSelectPage.tsx:47]: We probably need a more understandable function name 😄 
```suggestion
    const updateBankAccount = useCallback(
```

File: src/pages/workspace/accounting/xero/export/XeroBankAccountSelectPage.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        ({value}: SelectorType) => {
+            if (initiallyFocusedOptionKey !== value) {
+                Connections.updatePolicyConnectionConfig(policyID, CONST.POLICY.CONNECTIONS.NAME.XERO, CONST.XERO_CONFIG.EXPORT, {
+                    nonReimbursableAccount: value,
+                });
+            }
+            Navigation.goBack(ROUTES.POLICY_ACCOUNTING_XERO_ADVANCED.getRoute(policyID));
```
Comment (hungvu193) [inline src/pages/workspace/accounting/xero/export/XeroBankAccountSelectPage.tsx:54]:
```suggestion
            Navigation.goBack(ROUTES.POLICY_ACCOUNTING_XERO_EXPORT.getRoute(policyID));
```

**Review by hungvu193 — CHANGES_REQUESTED** [review]
Looking good. Only 2 small comments 👍

**Comment by mananjadhav** [comment]
Thanks for the important redirect bug. I had my commits messed up, could've easily missed this one.

**Review by hungvu193 — APPROVED** [review]
LGTM 😄

**Comment by mananjadhav** [comment]
@hungvu193 @ishpaul777 Can one of you please complete the checklist for this one?

**Review by ishpaul777 — APPROVED** [review]

**Review by pecanoro — APPROVED** [review]

**Comment by lakchote** [comment]
@mananjadhav there are conflicts that need to be solved

**Review by lakchote — APPROVED** [review]

**Comment by mananjadhav** [comment]
Running prettier diff.

**Comment by mananjadhav** [comment]
Fixed lint issues. All checks should pass now.

**Pull request merged**